CMakePresets.json

```diff
@@ -0,0 +1,173 @@
+{
+  "version": 3,
+  "configurePresets": [
+    {
+      "name": "base",
+      "displayName": "base preset",
+      "generator": "Ninja",
+      "binaryDir": "${sourceDir}/build/${presetName}",
+      "cacheVariables": {
+        "CMAKE_EXPORT_COMPILE_COMMANDS": true
+      },
+      "environment": {
+        "LOG_DIR": "${sourceDir}/build/${presetName}/logs"
+      },
+      "hidden": true
+    },
+    {
+      "name": "release",
+      "displayName": "Release",
+      "description": "Same as RelWithDebInfo, but disables debug information",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "Release"
+      },
+      "inherits": [
+        "base"
+      ]
+    },
+    {
+      "name": "relwithdebinfo",
+      "displayName": "RelWithDebInfo",
+      "description": "Enables optimizations (-Og or -O2) with debug information",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "RelWithDebInfo"
+      },
+      "inherits": [
+        "base"
+      ]
+    },
+    {
+      "name": "debug",
+      "displayName": "Debug",
+      "description": "Disables optimizations (-O0), enables debug information",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "Debug"
+      },
+      "inherits": [
+        "base"
+      ]
+    },
+    {
+      "name": "debug-clang",
+      "displayName": "Debug (w/ clang toolchain)",
+      "environment": {
+        "CC": "clang",
+        "CXX": "clang++"
+      },
+      "inherits": [
+        "debug"
+      ]
+    },
+    {
+      "name": "asan",
+      "displayName": "Debug (w/ ASAN/UBSAN support)",
+      "cacheVariables": {
+        "CLANG_ASAN_UBSAN": "ON"
+      },
+      "environment": {
+        "ASAN_OPTIONS": "'detect_leaks=0;log_path=$env{LOG_DIR}'",
+        "UBSAN_OPTIONS": "'print_stacktrace=1'"
+      },
+      "inherits": [
+        "debug-clang"
+      ]
+    },
+    {
+      "name": "tsan",
+      "displayName": "Debug (w/ TSAN support)",
+      "cacheVariables": {
+        "CLANG_TSAN": "ON"
+      },
+      "environment": {
+        "TSAN_OPTIONS": "'log_path=$env{LOG_DIR}'"
+      },
+      "inherits": [
+        "debug-clang"
+      ]
+    },
+    {
+      "name": "msan",
+      "displayName": "Debug (w/ MSAN support)",
+      "cacheVariables": {
+        "CLANG_MSAN": "ON"
+      },
+      "environment": {
+        "MSAN_OPTIONS": "'log_path=$env{LOG_DIR}'"
+      },
+      "inherits": [
+        "debug-clang"
+      ]
+    },
+    {
+      "name": "win-x64",
```

Comment:
just `win` is fine.

Reply:
the problem is that by default, CMake will still use x86 as the default target for VS 2017 and older, so I figured it's good to make this obvious by both forcing a platform target which is also visible in the name of the preset, see https://cmake.org/cmake/help/latest/variable/CMAKE_VS_PLATFORM_NAME_DEFAULT.html